server/chat/background/task.py

```
@@ -37,6 +37,34 @@
 logger = logging.getLogger(__name__)
 
 
+# ---------------------------------------------------------------------------
+# Per-worker Agent singleton: avoids recreating PostgreSQLClient, WeaviateClient,
+# and LLMManager on every background chat task (~2s cold-start savings).
+# ---------------------------------------------------------------------------
+_worker_agent = None
+
+
+def _get_worker_agent():
```

Ruff (0.15.17)

[warning] Missing return type annotation for private function `_get_worker_agent`

(ANN202)